examples/flower.rs

```diff
@@ -0,0 +1,110 @@
+//! Draws a simple geometric sort of flower with customizable dimensions.
+//!
+//! This example makes extensive use of the turtle arc methods: [`arc_left`] and [`arc_right`].
+//! They both provide the ability to draw a circular arc defined by the given `radius` and `extent`
+//! parameters, while the first makes the turtle draw it to the left and the second to the right.
+
+use turtle::{Angle, Distance, Drawing};
+
+const TURTLE_SPEED: &str = "faster";
+const BOTTOM_MARGIN: Distance = 25.0;
+
+const LEAF_FILL_COLOR: &str = "green";
+const LEAF_BORDER_COLOR: &str = "dark green";
+const LEAF_BORDER_WIDTH: Distance = 1.0;
+const LEFT_LEAF_RADIUS: Distance = 200.0;
+const LEFT_LEAF_EXTENT: Angle = 45.0;
+const RIGHT_LEAF_INCLINATION: Angle = 15.0;
+const RIGHT_LEAF_BOTTOM_RADIUS: Distance = 250.0;
+const RIGHT_LEAF_BOTTOM_EXTENT: Angle = 45.0;
+const RIGHT_LEAF_TOP_RADIUS: Distance = 157.0;
+const RIGHT_LEAF_TOP_EXTENT: Angle = 75.0;
+
+const TRUNK_COLOR: &str = LEAF_BORDER_COLOR;
+const TRUNK_WIDTH: Distance = 3.0;
+const TRUNK_PIECE_COUNT: usize = 4;
+const TRUNK_PIECE_RADIUS: Distance = 500.0;
+const TRUNK_PIECE_EXTENT: Angle = 15.0;
+
+const PETALS_COUNT: usize = 4;
+const PETALS_FILL_COLOR: &str = "purple";
+const PETALS_BORDER_COLOR: &str = "dark purple";
+const PETALS_BORDER_WIDTH: Distance = LEAF_BORDER_WIDTH;
+const PETALS_INIT_LEFT: Angle = 65.0;
+const PETALS_SIDE_RADIUS: Distance = 80.0;
+const PETALS_SIDE_EXTENT: Angle = 90.0;
+const PETALS_SPACE_GAP: Angle = 20.0;
+const PETALS_SPACE_RADIUS: Distance = 40.0;
+const PETALS_SPACE_EXTENT: Angle = 30.0;
+
+fn main() {
+    // Acquiring resources.
+    let mut drawing = Drawing::new();
+    let size = drawing.size();
+    let mut turtle = drawing.add_turtle();
+
+    // Initial positioning.
+    turtle.set_speed("instant");
+    turtle.pen_up();
+    turtle.go_to([
+        -(size.width as f64) / 6.0,
+        -(size.height as f64) / 2.0 + BOTTOM_MARGIN,
+    ]);
+    turtle.pen_down();
+
+    // Setup.
+    turtle.use_degrees();
+    turtle.set_speed(TURTLE_SPEED);
+
+    // Body.
+    turtle.set_fill_color(LEAF_FILL_COLOR);
+    turtle.set_pen_color(LEAF_BORDER_COLOR);
+
+    for _ in 0..TRUNK_PIECE_COUNT {
+        // Leaves:
+        turtle.set_pen_size(LEAF_BORDER_WIDTH);
+        turtle.set_pen_color(LEAF_BORDER_COLOR);
+        turtle.begin_fill();
+
+        // Left leaf.
+        turtle.arc_left(LEFT_LEAF_RADIUS, LEFT_LEAF_EXTENT);
+        turtle.right(LEFT_LEAF_EXTENT);
+        turtle.arc_right(LEFT_LEAF_RADIUS, -LEFT_LEAF_EXTENT);
+        turtle.right(LEFT_LEAF_EXTENT);
+
+        // Right leaf.
+        turtle.right(RIGHT_LEAF_INCLINATION);
+        turtle.arc_right(RIGHT_LEAF_BOTTOM_RADIUS, RIGHT_LEAF_BOTTOM_EXTENT);
+        turtle.right(RIGHT_LEAF_INCLINATION);
+        turtle.arc_right(RIGHT_LEAF_TOP_RADIUS, -RIGHT_LEAF_TOP_EXTENT);
```

Comment:
I don't really see the point in doing this, because it just complicates the reading in my opinion. What I think would be more interesting is explaining why I have used negative extents in order to go backwards and avoid having to U-turn regularly. We could at the same time explain that a negative radius means exchanging `arc_left` and `arc_right`, but I don't feel comfortable with actually putting it in.

Reply:
Keep in mind that this code is for illustration purposes only, so it's okay to put in something illustrative like this even though you'd probably never do that in real life. The point is to demonstrate the effect of the negative radius, even though it's likely that you'd probably prefer to use the other method here.

If you can find a better place for it I am totally fine with that. I just want to make sure we have the negative radius case covered *somewhere*.